lib/store/index.js

```diff
@@ -70,6 +72,27 @@ class Store {
 
     this.grammar = grammar;
   }
+
+  @action setBusySignalRegistry(registry) {
+    this.busySignalRegistry = registry;
+    this.runningKernels.forEach(this._registerBusySignal);
+  }
+
+  _registerBusySignal(kernel: Kernel) {
```

Comment:
As I find a bit confusing that the last message in the history is `kernel is busy`, how about something like this?

```js
  _registerBusySignal(kernel: Kernel) {
    if (this.busySignalRegistry) {
      const busySignalProvider = this.busySignalRegistry.create();
      reaction(
        () => kernel.executionState,
        state => {
          const message = kernel.displayName + " is " + state;
          busySignalProvider.add(message);
          if (state === "idle") busySignalProvider.clear();
        }
      );
    }
  }
```